Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -420,13 +420,13 @@ extension TabTrayController: TabManagerDelegate {
 
     func tabManager(tabManager: TabManager, didRemoveTab tab: Browser, atIndex index: Int) {
         var newTab: Browser? = nil
-        self.collectionView.performBatchUpdates({ _ in
-            self.collectionView.deleteItemsAtIndexPaths([NSIndexPath(forItem: index, inSection: 0)])
-        }, completion: { finished in
-            if tabManager.count == 0 {
-                newTab = tabManager.addTab()
-            }
-        })
+
+        self.collectionView.deleteItemsAtIndexPaths([NSIndexPath(forItem: index, inSection: 0)])
+        self.collectionView.reloadItemsAtIndexPaths(self.collectionView.indexPathsForVisibleItems())
+
+        if tabManager.count == 0 {
+            newTab = tabManager.addTab()
```

Comment:
I agree. I moved this into the tabManager but then found that it caused a bug with updating the tab counter. The BVC and Tab Tray are both delegates and depending on the order of which they get called the count can result in 0. I've added a fix but always setting the count to at least one so we never set the tab count to zero.